components/courses/app/services/courses/journal/total.rb

```diff
@@ -0,0 +1,45 @@
+module Courses
+  module Journal
+    class Total < ApplicationService
+      def initialize(student)
+        @student   = student
+      end
+
+      def call
+        total
+      end
+
+      private
+
+      attr_reader :student
+
+      def marks_total
+        sum = 0
+        progresses.each do |progress|
```

Comment:
`progresses.map(&: homework_mark).compact` and then no unless need... you know what to do